web/src/main/kotlin/com/example/ajaxproject/service/GroupChatServiceImpl.kt

```diff
@@ -30,107 +34,94 @@ class GroupChatServiceImpl (
         private val logger: Logger = LoggerFactory.getLogger(GroupChatServiceImpl::class.java)
     }
 
-    override fun createGroupRoom(createChatDto: CreateChatDto): GroupChatRoom {
-
-        val user = userService.getUserById(createChatDto.adminId)
-
-        val groupChatRoom = GroupChatRoom(
-            id = ObjectId().toHexString(),
-            chatName = createChatDto.chatName,
-            adminId = user.id,
-            chatMembers = listOf(user)
-        )
-
-        groupChatRoom.chatMembers + user
-
-        groupChatRoomRepository.save(groupChatRoom)
-
-        logger.info("Group chat room created successfully")
-
-        return groupChatRoom
-
+    override fun createGroupRoom(createChatDto: CreateChatDTO): Mono<GroupChatRoom> {
+        return userService.getById(createChatDto.adminId)
+            .flatMap { user ->
+                val groupChatRoom = GroupChatRoom(
+                    id = ObjectId().toHexString(),
+                    chatName = createChatDto.chatName,
+                    adminId = user.id,
+                    chatMembers = listOf(user)
+                )
+                groupChatRoomRepository.save(groupChatRoom)
+                    .doOnSuccess { logger.info("Group chat room created successfully") }
+            }
     }
 
-    override fun getAllChatMembers(chatId: String): List<User> {
-
-        val chatMembers = groupChatRoomRepository.findChatRoom(chatId).chatMembers
-
-        logger.info("Retrieved chat members for chat ID: {}" , chatId)
-
-        return chatMembers
+    override fun getAllChatMembers(chatId: String): Mono<List<User>> {
+        return groupChatRoomRepository.findChatRoom(chatId)
+            .flatMapMany { chatRoom ->
+                Flux.fromIterable(chatRoom.chatMembers)
+            }
+            .collectList().doOnSuccess { _ -> logger.info("Retrieved chat members for chat ID: {}", chatId) }
     }
 
-    override fun addUserToChat(chatId: String, userId: String): List<User> {
-
-        val chat = groupChatRoomRepository.findChatRoom(chatId)
-        val user = userService.getUserById(userId)
-
-        val existingUser = chat.chatMembers.find { it.id == userId }
 
-        if (existingUser != null) {
-            logger.info("User with ID {} is already present in chat ID {}", userId, chatId)
-            throw WrongActionException("User is already present in this chat")
+    override fun addUserToChat(chatId: String, userId: String): Mono<User> {
+        return Mono.zip(
+            groupChatRoomRepository.findChatRoom(chatId),
+            userService.getById(userId)
+        ).flatMap { (chat, user) ->
+            val existingUser = chat.chatMembers.find { it.id == userId }
+            if (existingUser != null) {
+                logger.info("User with ID {} is already present in chat ID {}", userId, chat)
```

Comment:
```suggestion
                logger.info("User with ID {} is already present in chat ID {}", userId, chatId)
```
Your log message contains `chat ID` but you pass entire `chat` object

Reply:
Agree

Reply:
I agree